### PR TITLE
Use AvgPool-14 in PT FE

### PR DESCRIPTION
**Pull request by p-wysocki**
### Details:
- Extend PT FE with AvgPool-14
 - Move the `count_include_pad` workaround subgraph from PT FE to a downgrade transformation
 - The subgraph in downgrade transformations is defined in https://github.com/openvinotoolkit/openvino/blob/master/src/common/transformations/src/transformations/op_conversions/convert_avgpool_downgrade.cpp#L52
 - The new `RoundingType::CEIL_TORCH` has impact only on the output shape
 - Out of two tests which previously have been xfailed due to output shape mismatch one is still failing, but now due to accuracy validation failure
 - No old tests were broken by upgrading to V14, so it seems the subgraph has issue with this edge case specifically
 - The edge case is connected to `count_include_pad:True`, and `pad` value, depending on the latter FP16 test case sometimes passes, but FP32 always fails
 - Since CPU Plugin is calling oneDNN AvgPool implementation there may be some difference in results calculation or an issue with parameter preprocessing before calling oneDNN function
 - Continued in XXXXX

### Tickets:
 - CVS-133929

### CODE REVIEW

**Review by praasz — APPROVED** [review]

**Comment by mlukasze** [comment]
setting `do not merge` label temporary, as we found additional issue with operator and it's need to be discussed before we enable it for the frontend

**Comment by mvafin** [comment]
With merge of MaxPool-14 we had performance regression on CPU before it was implemented in plugin and on GPU. Can we test that this change doesn't degrade performance?

File: tests/layer_tests/pytorch_tests/test_pooling.py [hunk 1]
```diff
@@ -161,6 +153,8 @@ def test_avg_pool1d(self, params, ceil_mode, count_include_pad, ie_device, preci
     @pytest.mark.xfail(condition=platform.system() == 'Darwin' and platform.machine() == 'arm64',
                        reason='Ticket - 122715')
     def test_avg_pool2d(self, params, ceil_mode, count_include_pad, ie_device, precision, ir_version):
+        if ceil_mode and count_include_pad and np.array_equal(np.array(params["kernel_size"]), np.array([8, 8])):
+            pytest.skip("ticket XXXXX")
```
Comment (mvafin) [inline tests/layer_tests/pytorch_tests/test_pooling.py:157]: Please mark it as xfail

Comment (p-wysocki) [reply tests/layer_tests/pytorch_tests/test_pooling.py]: Done.

**Review by mvafin — APPROVED** [review]

**Comment by eaidova** [comment]
@p-wysocki please make sure that updasted IR works on all devices including NPU and downgrade transformation where op is not implemented applied. We still see issue for maxpool (during 2 releases for NPU), not lower amount computer vision models will be affected by switching by avgpool

**Pull request merged**